lib/mcb.rb

```diff
@@ -195,6 +195,38 @@ def apiv1_opts(opts)
       opts
     end
 
+    # Return options necessary to connect to API V2.
+    #
+    # The opts passed in are examined determine which opts need to be added,
+    # this function essentially just fills in any missing options.
+    #
+    #   opts = apiv2_opts(opts)
+    def apiv2_opts(opts)
+      opts[:url] = opts[:url]
+      opts[:'max-pages'] = opts[:'max-pages']
+      opts[:token] = opts[:token]
+      opts[:all] = opts[:all]
+
+      opts.merge! azure_env_settings_for_opts(**opts)
+
+      if requesting_remote_connection?(**opts)
+        opts[:url] = MCB::Azure.get_urls(**opts).first
+        opts[:token] = MCB::Azure.get_config(**opts)['AUTHENTICATION_TOKEN']
```

Comment:
It seems confusing to conditionally overwrite the url and token, I think it'd be better if we assigned one or the other.